chain/messagepool/messagepool.go

```diff
@@ -355,12 +355,34 @@ func (mp *MessagePool) addLocal(m *types.SignedMessage, msgb []byte) error {
 	return nil
 }
 
-func (mp *MessagePool) verifyMsgBeforeAdd(m *types.SignedMessage, epoch abi.ChainEpoch) error {
+func (mp *MessagePool) verifyMsgBeforeAdd(m *types.SignedMessage, curTs *types.TipSet, local bool) error {
+	epoch := curTs.Height()
 	minGas := vm.PricelistByEpoch(epoch).OnChainMessage(m.ChainLength())
 
 	if err := m.VMMessage().ValidForBlockInclusion(minGas.Total()); err != nil {
 		return xerrors.Errorf("message will not be included in a block: %w", err)
 	}
+
+	// this checks if the GasFeeCap is suffisciently high for inclusion in the next 20 blocks
+	// if the GasFeeCap is too low, we soft reject the message (Ignore in pubsub) and rely
+	// on republish to push it through later, if the baseFee has fallen.
+	// this is a defensive check that stops minimum baseFee spam attacks from overloading validation
+	// queues.
+	// Note that we don't do that for local messages, so that they can be accepted and republished
+	// automatically
+	if !local {
+		baseFee, err := mp.api.ChainComputeBaseFee(context.TODO(), curTs)
```

Comment:
Looking at the computed base fee is much more expensive then just looking at base fee in the curTs.
`curTs.Blocks()[0].ParentBaseFee`

Reply:
ah, good point; will do this.

Reply:
done!